packages/libp2p/test/dcutr/dcutr.node.ts

```diff
@@ -164,9 +185,22 @@ describe('dcutr', () => {
       const relayedAddress = multiaddr(`/ip4/127.0.0.1/tcp/${RELAY_PORT}/p2p/${relay.peerId}/p2p-circuit/p2p/${libp2pB.peerId}`)
       const connection = await libp2pA.dial(relayedAddress)
 
+      log(`initial connection ID: ${connection.id}`)
+
       // connection should be transient
       expect(connection).to.have.property('transient', true)
 
+      // wait for the relayedAddress connection to be closed
+      const closedRelayConnection = async () => await pRetry(async () => {
+        if (connection.status !== 'closed') {
+          throw new Error('relayed connection is still open')
+        }
+        return true
+      }, {
+        retries: 10
+      })
+      expect(closedRelayConnection()).to.eventually.equal(true)
```

Comment:
we could add a delay here, but this allows us to assert more directly that the relayConnection is closed

Reply:
You need to await anything with `.eventually` as it returns a promise:

```suggestion
      // wait for the relayedAddress connection to be closed
      const closedRelayConnection = async () => await pRetry(async () => {
        if (connection.status !== 'closed') {
          throw new Error('relayed connection is still open')
        }
        return true
      }, {
        retries: 10
      })
      await expect(closedRelayConnection()).to.eventually.equal(true)
```